src/psyclone/dynamo0p3.py

```diff
@@ -5956,19 +5956,24 @@ def _compute_halo_depth(self):
         :rtype: str
 
         '''
-        # get information about reading from the halo from all read fields
+        # Get information about reading from the halo from all "read" fields
         # dependendent on this halo exchange
         depth_info_list = self._compute_halo_read_depth_info()
 
-        # if there is only one entry in the list we can just return
-        # the depth
-        if len(depth_info_list) == 1:
-            return str(depth_info_list[0])
-        # the depth information can't be reduced to a single
-        # expression, therefore we need to determine the maximum
-        # of all expresssions
+        # Create string representation of depth info list and filter
```

Comment:
I'm afraid I don't like this fix - it's fixing the symptom rather than the underlying disease. I've done a bit of digging and the problem seems to be that one of the HaloReadAccess objects returned by `_compute_halo_read_info` has a literal depth of zero and no other depth information (i.e. var_depth is None and `max_depth`, `max_depth_m1` and `annexed_only` are all `False`). I suspect this should never happen. If I were you @TeranIvy I'd pass this on to @rupertford now that you've created a reproducer :-)